addons/website_slides/models/slide_channel.py

```diff
@@ -738,7 +743,24 @@ def _remove_membership(self, partner_ids):
         self.message_unsubscribe(partner_ids=partner_ids)
 
         if removed_channel_partner_domain:
-            self.env['slide.channel.partner'].sudo().search(removed_channel_partner_domain).unlink()
+            self.env['slide.channel.partner'].sudo().search(removed_channel_partner_domain).write({
+                'active': False,
+            })
+
+    def _update_user_karma(self, partner_ids, joining=True):
+        """ Update the user karma while leaving / joining the course.
+
+            :param bool joining: If true, karma will added for the user(s),
+              otherwise (while leaving the course), it will be deducted.
+        """
+        earned_karma = self._get_earned_karma(partner_ids)
```

Comment:
Actually it was before it was used just to remove the member and subtract the karma.
But, we want to add a feature to add the previous karma again.
so that I have split the code and made it work like both Joining and Removing.

Thanks